src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertificateAssetDownloader.cs

```diff
@@ -130,55 +130,118 @@ internal static class CertificateAssetDownloader
+                PropertyInfo? allowAutoRedirectProp = socketsHttpHandlerType.GetProperty("AllowAutoRedirect");
+                ConstructorInfo? httpClientCtor = httpClientType.GetConstructor(new Type[] { httpMessageHandlerType });
                 PropertyInfo? requestUriProp = httpRequestMessageType.GetProperty("RequestUri");
                 ConstructorInfo? httpRequestMessageCtor = httpRequestMessageType.GetConstructor(Type.EmptyTypes);
                 MethodInfo? sendMethod = httpClientType.GetMethod("Send", new Type[] { httpRequestMessageType, typeof(CancellationToken) });
                 PropertyInfo? responseContentProp = httpResponseMessageType.GetProperty("Content");
+                PropertyInfo? responseStatusCodeProp = httpResponseMessageType.GetProperty("StatusCode");
+                PropertyInfo? responseHeadersProp = httpResponseMessageType.GetProperty("Headers");
+                PropertyInfo? responseHeadersLocationProp = httpResponseHeadersType.GetProperty("Location");
                 MethodInfo? readAsStreamMethod = httpContentType.GetMethod("ReadAsStream", Type.EmptyTypes);
-                if (pooledConnectionIdleTimeoutProp == null || requestUriProp == null || httpRequestMessageCtor == null || sendMethod == null || responseContentProp == null || readAsStreamMethod == null)
+
+                if (socketsHttpHandlerCtor == null || pooledConnectionIdleTimeoutProp == null || allowAutoRedirectProp == null || httpClientCtor == null ||
+                    requestUriProp == null || httpRequestMessageCtor == null || sendMethod == null || responseContentProp == null || responseStatusCodeProp == null ||
+                    responseHeadersProp == null || responseHeadersLocationProp == null || readAsStreamMethod == null)
                 {
                     Debug.Fail("Unable to load required member.");
                     return null;
                 }
 
                 // Only keep idle connections around briefly, as a compromise between resource leakage and port exhaustion.
                 const int PooledConnectionIdleTimeoutSeconds = 15;
+                const int MaxRedirections = 10;
 
                 // Equivalent of:
-                // var socketsHttpHandler = new SocketsHttpHandler() { PooledConnectionIdleTimeout = TimeSpan.FromSeconds(PooledConnectionIdleTimeoutSeconds) };
+                // var socketsHttpHandler = new SocketsHttpHandler() {
+                //     PooledConnectionIdleTimeout = TimeSpan.FromSeconds(PooledConnectionIdleTimeoutSeconds),
+                //     AllowAutoRedirect = false
+                // };
                 // var httpClient = new HttpClient(socketsHttpHandler);
-                object? socketsHttpHandler = Activator.CreateInstance(socketsHttpHandlerType);
+                // Note: using a ConstructorInfo instead of Activator.CreateInstance, so the ILLinker can see the usage through the lambda method.
```

Comment:
I don't think this comment makes sense. This isn't being called inside the lambda method. It should be fine to continue calling `Activator.CreateInstance` here.